example/pkg/apis/olympus/v1beta1/poseidon_types.go

```diff
@@ -101,6 +103,52 @@ func (b PoseidonStrategy) BasicMatch(label labels.Selector, field fields.Selecto
 	}
 }
 
+//ConvertToTable Starting with Kubernetes 1.11, kubectl uses server-side printing. The server decides which columns are shown by the kubectl get command. You can customize these columns by Implementing ConvertToTable
+func (PoseidonStrategy) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1beta1.Table, error) {
+	var table metav1beta1.Table
+	var swaggerMetadataDescriptions = metav1.ObjectMeta{}.SwaggerDoc()
+	table.ColumnDefinitions = []metav1beta1.TableColumnDefinition{
+		{Name: "Name", Type: "string", Format: "name", Description: metav1.ObjectMeta{}.SwaggerDoc()["name"]},
+		{Name: "Created At", Type: "date", Description: swaggerMetadataDescriptions["creationTimestamp"]},
+		{Name: "Namespace", Type: "string", Description: "Namespace"},
+	}
+	fn := func(obj runtime.Object) error {
+		m, err := meta.Accessor(obj)
+		if err != nil {
+			return fmt.Errorf("the resource %s does not support server-side printing", "Poseidon")
+		}
+		table.Rows = append(table.Rows, metav1beta1.TableRow{
+			Cells: []interface{}{
+				m.GetName(),
+				m.GetCreationTimestamp().Time.UTC().Format(time.RFC3339),
+				m.GetNamespace()},
+			Object: runtime.RawExtension{Object: obj},
+		})
+		return nil
+	}
+	switch {
+	case meta.IsListType(object):
+		if err := meta.EachListItem(object, fn); err != nil {
```

Comment:
@yue9944882 this line uses `meta.EachListItem`function, its second parameter is a `func(obj runtime.Object) error`. So the `fn` is function that accept `object` instead of a true type.

Reply:
> @yue9944882 this line uses meta.EachListItemfunction, its second parameter is a func(obj runtime.Object) error. So the fn is function that accept object instead of a true type.

wdym? printers should accept hub types. e.g.

[https://github.com/kubernetes/kubernetes/blob/1c04caa04325e1f64d9a15714ad61acdd2a81013/pkg/printers/internalversion/printers.go#L91-L92](https://github.com/kubernetes/kubernetes/blob/1c04caa04325e1f64d9a15714ad61acdd2a81013/pkg/printers/internalversion/printers.go#L91-L92)

Reply:
Perhaps i should ask you what is `hub type`. I would have misunderstand your words. -o-

Reply:
> Perhaps i should firstly ask you what is hub type. I would have misunderstand your words. -o- Is it the struct under the interface?

> should only work w/ unversioned/hub type in the storage.

hub/unversioned/internal/handler/runtime types is the only type processed by its registry